packages/search-input/src/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,408 @@
+        closeMenu();
+      }
+    };
+
+    const handleSubmit: FormEventHandler<HTMLFormElement> = e => {
+      e.preventDefault(); // prevent page reload
+      onSubmitProp?.(e);
+      if (withTypeAhead) {
+        closeMenu();
+      }
+    };
+
+    useBackdropClick(closeMenu, [searchBoxRef, menuRef], isOpen);
+
+    return (
+      <LeafyGreenProvider darkMode={darkMode}>
+        <SearchInputContextProvider
+          state={state}
+          highlight={highlightIndex}
+          resultDynamicRefs={resultRefs}
+        >
+          <form
+            role="search"
+            ref={formRef}
+            className={cx(formStyle, className)}
+            onSubmit={handleSubmit}
+            onBlur={handleBlur}
+            {...rest}
+          >
+            {/* Disable eslint: onClick sets focus. Key events would already have focus */}
+            {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions */}
+            <div
+              ref={searchBoxRef}
+              role="searchbox"
+              tabIndex={-1}
+              onMouseDown={handleSearchBoxMousedown}
+              onClick={handleSearchBoxClick}
+              onFocus={handleSearchBoxFocus}
+              onKeyDown={handleKeyDown}
+              className={cx(
+                inputWrapperStyle,
+                inputWrapperSizeStyle[size],
+                inputWrapperThemeStyle[theme],
+                {
+                  [inputWrapperFocusStyles[theme]]:
+                    focusedElement === inputRef.current,
+                  [inputWrapperDisabledStyle[theme]]: disabled,
+                  [inputWrapperInteractiveStyles[theme]]: !disabled,
+                },
+              )}
+              aria-label={ariaLabel}
+              aria-labelledby={ariaLabelledBy}
+            >
+              <MagnifyingGlass
+                className={cx(
+                  searchIconThemeStyle[theme],
+                  searchIconSizeStyle[size],
+                  { [searchIconDisabledStyle[theme]]: disabled },
+                )}
+                role="presentation"
+              />
+              <input
+                type="search"
+                className={cx(baseInputStyle, inputThemeStyle[theme])}
+                value={value}
+                onChange={handleChange}
+                placeholder={placeholder}
+                ref={inputRef}
+                disabled={disabled}
+              />
+              {value && (
+                <IconButton
+                  ref={clearButtonRef}
+                  type="button"
+                  aria-label="Clear search"
+                  onClick={handleClearButton}
+                  className={clearButtonSizeStyle[size]}
+                >
+                  <XIcon />
+                </IconButton>
+              )}
+            </div>
+            {withTypeAhead && (
+              <SearchResultsMenu
```

Comment:
Should we add `aria-describedby` here?

Reply:
What element would this be described by? 
If the user passes a `describedby` prop it'll be spread onto the `form` element which I believe is fine

Reply:
oh sorry, I meant to add this comment on the `input` above. I'm not sure if this applies since its a search input but if `<SearchResultsMenu>` had an id it could be passed to the `aria-describedby` on the input.